network/src/commonMain/kotlin/com/wire/kalium/network/session/SessionManager.kt

```diff
@@ -55,6 +58,7 @@ fun HttpClientConfig<*>.installAuth(sessionManager: SessionManager, accessTokenA
             }
 
             refreshTokens {
+                sessionManager.beforeTokenUpdate()
```

Comment:
Instead of doing:

```
 lock
 refresh token
 update token
 unlock 
 ```
 
 maybe we should move this whole logic to the sessionManager and only enqueue a token refresh here?

Reply:
I think using a queue we can avoid a lock

Reply:
A queue in Kotlin I guess is a dispatcher with parallelism of one.